lambdas/data-migration1/src/providers.ts

```diff
@@ -0,0 +1,157 @@
+import Knex from 'knex';
+import isNil from 'lodash/isNil';
+
+import DynamoDbSearchQueue from '@cumulus/aws-client/DynamoDbSearchQueue';
+import * as KMS from '@cumulus/aws-client/KMS';
+import { envUtils, keyPairProvider } from '@cumulus/common';
+import Logger from '@cumulus/logger';
+
+import { RecordAlreadyMigrated } from './errors';
+import { MigrationSummary } from './types';
+
+const Manager = require('@cumulus/api/models/base');
+const schemas = require('@cumulus/api/models/schemas');
+
+const logger = new Logger({ sender: '@cumulus/data-migration/providers' });
+
+export interface RDSProviderRecord {
+  name: string
+  protocol: string
+  host: string
+  port?: number
+  username?: string
+  password?: string
+  encrypted?: boolean
+  globalConnectionLimit?: number
+  privateKey?: string
+  cmKeyId?: string
+  certificateUri?: string
+  created_at: Date
+  updated_at?: Date
+}
+
+const encryptProviderCredential = async (
+  value: string,
+  encrypted?: boolean
+) => {
+  if (isNil(value)) return undefined;
+
+  const providerKmsKeyId = envUtils.getRequiredEnvVar('provider_kms_key_id');
+
+  if (encrypted) {
+    return keyPairProvider.S3KeyPairProvider
+      .decrypt(value)
+      .then(
+        (decryptedValue) => KMS.encrypt(providerKmsKeyId, decryptedValue),
+        async (error) => {
+          // If we already have a KMS encrypted value, return it.
+          if ((await KMS.decryptBase64String(value))) {
+            return value;
+          }
+          throw error;
+        }
+      );
+  }
+
+  return KMS.encrypt(providerKmsKeyId, value);
+};
+
+/**
+ * Migrate provider record from Dynamo to RDS.
+ *
+ * @param {AWS.DynamoDB.DocumentClient.AttributeMap} dynamoRecord
+ *   Source record from DynamoDB
+ * @param {Knex} knex - Knex client for writing to RDS database
+ * @returns {Promise<number>} - Cumulus ID for record
+ * @throws {RecordAlreadyMigrated}
+ *   if record was already migrated
+ */
+export const migrateProviderRecord = async (
+  dynamoRecord: AWS.DynamoDB.DocumentClient.AttributeMap,
```

Comment:
The answer may be "no", but is it worth creating a more provider-specific type for this record, rather than just accepting anything?

Reply:
Since we're immediately turning around and validating the dynamo record against the provider schema, does it matter?

Reply:
Probably not in this case.